src/wallet/salvage.cpp

```diff
@@ -20,6 +20,11 @@ bool RecoverDatabaseFile(const fs::path& file_path)
     std::string filename;
     std::shared_ptr<BerkeleyEnvironment> env = GetWalletEnv(file_path, filename);
 
+    if (!env->Open(true /* retry */)) {
+        tfm::format(std::cerr, "Error initializing wallet database environment %s!", env->Directory());
```

Comment:
in commit 17924a533346618531ed2c888c6e8fe67bc3851c:

Any reason to write to stderr when all other failures are written to the debug log? Since this is a pure utility function, I'd prefer if this returned the error string to the caller in the future, so that the caller can decide how to display the error

Reply:
It was to largely retain the previous behavior of this error going to stderr.